.github/workflows/statitc-coda-analysis.yml

```diff
@@ -0,0 +1,21 @@
+name: static-code-analysis
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+
+jobs:
+  black:
```

Comment:
A minor point, but I'd rename this to `lint` or `format`, without referencing a particular tool.